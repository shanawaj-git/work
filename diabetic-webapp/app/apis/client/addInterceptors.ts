import { APIClientOptions, AuthDataKeys } from '../types';

const storeAuthData = (user, jwt) => {
  sessionStorage.setItem(AuthDataKeys.User, JSON.stringify(user));
  sessionStorage.setItem(AuthDataKeys.Jwt, jwt);
};

const clearAuthData = () => {
  sessionStorage.removeItem(AuthDataKeys.User);
  sessionStorage.removeItem(AuthDataKeys.Jwt);
};

const getJwt = () => sessionStorage.getItem(AuthDataKeys.Jwt);

export const responseErrorInterceptor = options => error => {
  if (error.response.status === 401 || error.response.status === 403) {
    clearAuthData();
    options.onUnauthorizedAccess();
  }
  throw error;
};

export const responseInterceptor = options => response => {
  if (response.config.url === options.loginUrl && response.status === 200) {
    const { user, jwt } = response.data;
    storeAuthData(user, jwt);
    options.onSuccessfulAuthentication({
      user,
      jwt,
    });
  }
  return response;
};

export const requestInterceptor = config => {
  const headers = { ...config.headers };
  if (!config.isPublicUrl) {
    const jwt = getJwt();
    if (jwt) {
      headers.Authorization = `Bearer ${jwt}`;
    }
  }
  return { ...config, headers };
};
// Do something before request is sent

export const addInterceptors = (client, options: APIClientOptions) => {
  client.interceptors.response.use(
    responseInterceptor(options),
    responseErrorInterceptor(options),
  );
  client.interceptors.request.use(requestInterceptor);
};
