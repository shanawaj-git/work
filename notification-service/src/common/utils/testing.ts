import { MongooseModule, MongooseModuleOptions } from '@nestjs/mongoose';
import { MongoMemoryServer } from 'mongodb-memory-server';

let mongod: MongoMemoryServer;

export const rootMongooseTestModule = (options: MongooseModuleOptions = {}) =>
  MongooseModule.forRootAsync({
    useFactory: async () => {
      mongod = new MongoMemoryServer();
      await mongod.start();

      const mongoUri = mongod.getUri();
      return {
        uri: mongoUri,
        ...options,
      };
    },
  });

export const closeInMongodConnection = async () => {
  if (mongod) await mongod.stop();
};
