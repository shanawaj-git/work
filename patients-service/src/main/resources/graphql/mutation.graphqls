type Mutation{
    createPatient(input:PatientInput!):CreatePatientOutput!
    addPatientAddress(emiratesId: String!, input: AddressInput!): CreateAddressOutput
}