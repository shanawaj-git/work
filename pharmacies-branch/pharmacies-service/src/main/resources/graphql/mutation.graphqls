type Mutation{
    dummyMutation(input:String):String
}