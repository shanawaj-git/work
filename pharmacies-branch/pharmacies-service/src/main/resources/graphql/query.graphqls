type Query {
  dummyQuery: String
  getPharmacies( input: GetPharmaciesInput!): GetPharmaciesOutput!

}