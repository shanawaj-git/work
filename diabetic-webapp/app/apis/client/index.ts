import { APIClientOptions } from 'apis/types';
import axios from 'axios';
import { addInterceptors } from './addInterceptors';
import * as environmentConfig from '../../configs/environmentConfig';

let httpClient;
// eslint-disable-next-line no-unused-vars
const configureAPIClient = (options: APIClientOptions) => {
  // TODO: intercept response and add callback support for login, auth failed
  httpClient = axios.create({
    baseURL:
      environmentConfig.getEnvironmentConfig().API_BASE_URL ||
      'http://localhost:1337',
    timeout: 30000, // 30 seconds
    // TODO: pass Authorization once login is implemented
  });
  addInterceptors(httpClient, options);
  return httpClient;
};

const getAPIClientInstance = () => {
  if (!httpClient) {
    throw new Error('API Client is not intialized');
  }
  return httpClient;
};

export { configureAPIClient, getAPIClientInstance, APIClientOptions };
