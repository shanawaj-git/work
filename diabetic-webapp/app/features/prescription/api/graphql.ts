import { gql } from '@apollo/client';

export type GetPrescriptionInput = {
  prescriptionNumber: string;
};

export type GetPrescriptionOutput = {
  prescription: {
    success: boolean;
    data?: Prescription;
    error?: {
      code: PrescriptionErrorCode;
      message?: string;
    };
  };
};

export type Prescription = {
  prescriptionNumber: string;
  recordNumber?: string;
  visitDate?: Date;
  diagnosis?: string;
  pin?: string;

  insurance?: Insurance;

  drugs: Drug[];

  patient: Patient;
  doctor?: Doctor;
};

type Insurance = {
  insuranceId: string;
  policyNumber?: string;
  provider: InsuranceProvider;
};

type InsuranceProvider = {
  name: string;
};

type Patient = {
  name: Name;
  mohId?: string;
  emiratesId: string;
};

type Doctor = {
  name: Name;
};

type Name = {
  first: string;
  middle?: string;
  last: string;
};

export type Drug = {
  name: string;
  dosage?: Dosage;
};

type Dosage = {
  frequency?: string;
  unit?: string;
  period?: string;
  route?: string;
  quantity?: string;
  timeUnit?: string;
  doctorNotes?: string;
};

export const PrescriptionQueries = {
  Prescription: gql`
    query Prescription($input: GetPrescriptionInput!) {
      prescription(input: $input) {
        success
        data {
          prescriptionNumber
          recordNumber
          visitDate
          diagnosis
          pin

          patient {
            name {
              first
              middle
              last
            }
            mohId
            emiratesId
          }

          doctor {
            name {
              first
              middle
              last
            }
          }

          drugs {
            name

            dosage {
              frequency
              unit
              period
              route
              quantity
              timeUnit
              doctorNotes
            }
          }

          insurance {
            insuranceId
            policyNumber
            provider {
              name
            }
          }
        }

        error {
          code
          message
        }
      }
    }
  `,
};

export enum PrescriptionErrorCode {
  PrescriptionNotFound = 'PRESC_ERR_001',
}

export const OPERATION_GET_PRESCRIPTION = 'Prescription';
