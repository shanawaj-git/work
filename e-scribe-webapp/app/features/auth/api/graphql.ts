import { gql } from '@apollo/client';

export type GenerateOTPInput = {
  phoneNumber: string;
};

export type GenerateOTPOutput = {
  generateOTP: {
    success: boolean;
    error: {
      code: AuthErrorCode;
      message?: string;
      resendWaitPeriodMillis?: number;
    };
  };
};

export type ValidateOTPInput = {
  phoneNumber: string;
  otp: string,
};

export type ValidateOTPOutput = {
  validateOTP: {
    success: boolean;
    error: {
      code: AuthErrorCode;
      message?: string;
    };
  };
};

export const AuthMutations = {
  GenerateOTP: gql`
    mutation GenerateAuthOTP($input: GenerateOTPInput!) {
      generateOTP(input: $input) {
        success
        error {
          code
          message
        }
      }
    }
  `,
  ValidateOTP: gql`
  mutation ValidateAuthOTP($input: ValidateOTPInput!) {
    validateOTP(input: $input) {
      success
      error {
        code
        message
      }
    }
  }
`,
};

export enum AuthErrorCode {
  AwaitingResendDelay = 'AUTH_ERR_001',
  MaxAttemptsExceeded = 'AUTH_ERR_002',
  InvalidCredentials = 'AUTH_ERR_101',
  ExpiredOTP = 'AUTH_ERR_102',
};

export const OPERATION_VALIDATE_AUTH_OTP = 'ValidateAuthOTP';
