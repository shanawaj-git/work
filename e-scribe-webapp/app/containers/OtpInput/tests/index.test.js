import React from 'react';
import { render, fireEvent, getByTestId, wait } from 'react-testing-library';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { MockedProvider } from '@apollo/client/testing';
import { act } from 'react-test-renderer';

import appConfigs from 'configs/applicationConfig.json';
import { AuthErrorCode, AuthMutations } from 'features/auth/api/graphql';
import OTPInput from '../index';

const { OTP_LENGTH, RESEND_OTP_DELAY_IN_SECONDS } = appConfigs;

describe('<OTPInput />', () => {
  // eslint-disable-next-line no-global-assign
  window = Object.assign(window, {
    visualViewport: {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    },
  });

  const phoneNumber = '0123456789';
  const referrer = '/prescriptions/123456798';
  const mockLocation = {
    state: {
      phoneNumber,
      referrer,
    },
  };

  const otpVerficationInput = { phoneNumber, otp: '123456' };

  let mockHistory;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
  });

  afterEach(() => {
    mockHistory = null;
  });

  // eslint-disable-next-line react/prop-types
  const MockOtpInput = ({ locale }) => (
    <BrowserRouter>
      <IntlProvider locale={locale}>
        <MockedProvider>
          <OTPInput location={mockLocation} />
        </MockedProvider>
      </IntlProvider>
    </BrowserRouter>
  );

  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<MockOtpInput locale="en" />);
    expect(firstChild).toMatchSnapshot();
  });

  it('redirects to phone-login screen when phoneNumber is missing in the route params', () => {
    const OtpInputCmp = () => (
      <IntlProvider locale="en">
        <MockedProvider>
          <OTPInput
            location={{
              ...mockLocation,
              state: {
                phoneNumber: '',
              },
            }}
          />
        </MockedProvider>
      </IntlProvider>
    );

    const dummyPhoneLoginContent = 'Dummy phone login page content';
    const PhoneLoginCmp = () => dummyPhoneLoginContent;

    const { container } = render(
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={OtpInputCmp} />
          <Route exact path="/phone-login" component={PhoneLoginCmp} />
        </Switch>
      </BrowserRouter>,
    );

    expect(container.textContent).toBe(dummyPhoneLoginContent);
  });

  it(`OTP input length should be a fixed length (${OTP_LENGTH})`, () => {
    act(() => {
      render(<MockOtpInput locale="en" />);
    });
    const input = document.querySelector('.MuiInputBase-input');
    const inputNumber = '123456';
    fireEvent.change(input, { target: { value: `${inputNumber}` } });
    expect(input.value).toEqual(`${inputNumber}`);
  });
  it(`OPT input should not have more than ${OTP_LENGTH} digits`, () => {
    act(() => {
      render(<MockOtpInput locale="en" />);
    });
    const input = document.querySelector('.MuiInputBase-input');
    const inputNumber = '12345678';
    fireEvent.change(input, { target: { value: `${inputNumber}` } });
    expect(input.value).toEqual('');
  });
  it(`OTP input  should be numbers Only`, () => {
    act(() => {
      render(<MockOtpInput locale="en" />);
    });
    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, { target: { value: `hi i am number` } });
    expect(input.value).toBe('');
  });
  it('Timer should count down and localstorage should updated', () => {
    act(() => {
      render(<MockOtpInput locale="en" />);
    });
    const localStorageCounterVal = window.localStorage.getItem(
      'otp.resend.seconds.remaining',
    );
    expect(localStorageCounterVal).toBe(RESEND_OTP_DELAY_IN_SECONDS);
  });
  it('Button should be disabled before input change and OTP length matches', () => {
    act(() => {
      render(<MockOtpInput locale="en" />);
    });
    const nextBtn = document.querySelector('[data-testid="nextButton"]');
    expect(nextBtn.getAttribute('disabled')).toEqual('');
    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, { target: { value: '123456' } });
    expect(nextBtn.getAttribute('disabled')).toBeNull();
  });
  it(`App loads with initial counter of ${RESEND_OTP_DELAY_IN_SECONDS}`, () => {
    // eslint-disable-next-line no-shadow
    const { container } = render(<MockOtpInput locale="en" />);
    const countValue = getByTestId(container, 'counter-value');
    expect(countValue.textContent).toBe(
      `Resend in ${RESEND_OTP_DELAY_IN_SECONDS}s`,
    );
  });
  it('Timer Rest', async () => {
    render(<MockOtpInput locale="en" />);
    jest.runAllTimers();
    const resendLink = document.querySelector('.resend-link-text');
    expect(resendLink.textContent).toBe(`Resend`);
  });

  it('restarts the timer after a successful OTP is resent', async () => {
    const otpInput = { phoneNumber };
    const mocks = [
      {
        request: {
          query: AuthMutations.GenerateOTP,
          variables: { input: otpInput },
        },
        result: { data: { generateOTP: { success: true, error: null } } },
      },
    ];

    render(
      <BrowserRouter>
        <IntlProvider locale="en">
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );
    jest.runAllTimers();
    const resendLink = document.querySelector('[data-testid="resend-link"]');
    fireEvent.click(
      resendLink,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      const counterAfterTimeEllapsed = document.querySelector(
        '[data-testid="counter-value"]',
      );
      expect(counterAfterTimeEllapsed.textContent).toBe(
        `Resend in ${RESEND_OTP_DELAY_IN_SECONDS}s`,
      );
    });
  });

  it('displays error and resend option is enabled after resend otp failed with a specific reason', async () => {
    const otpInput = { phoneNumber };
    const mocks = [
      {
        request: {
          query: AuthMutations.GenerateOTP,
          variables: { input: otpInput },
        },
        result: {
          data: {
            generateOTP: {
              success: false,
              error: { code: AuthErrorCode.MaxAttemptsExceeded },
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.error.maxAttemptsExceeded';

    const messages = {
      [errorMsgCode]: 'Some error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );
    jest.runAllTimers();
    const resendLink = document.querySelector('[data-testid="resend-link"]');
    fireEvent.click(
      resendLink,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="resend-error-text"]').textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));

      expect(document.querySelector('.resend-link-text').textContent).toBe(
        `Resend`,
      );
    });
  });

  it('displays the generic error and resend option is enabled after resend otp failed with an unknown reason', async () => {
    const otpInput = { phoneNumber };
    const mocks = [
      {
        request: {
          query: AuthMutations.GenerateOTP,
          variables: { input: otpInput },
        },
        result: {
          data: {
            generateOTP: {
              success: false,
              error: { code: 'auth.err.unknownReason' },
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );
    jest.runAllTimers();
    const resendLink = document.querySelector('[data-testid="resend-link"]');
    fireEvent.click(
      resendLink,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="resend-error-text"]').textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));

      expect(document.querySelector('.resend-link-text').textContent).toBe(
        `Resend`,
      );
    });
  });

  it('displays the generic error and resend option is enabled after resend otp failed with missing error code', async () => {
    const otpInput = { phoneNumber };
    const mocks = [
      {
        request: {
          query: AuthMutations.GenerateOTP,
          variables: { input: otpInput },
        },
        result: {
          data: {
            generateOTP: {
              success: false,
              error: {},
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );
    jest.runAllTimers();
    const resendLink = document.querySelector('[data-testid="resend-link"]');
    fireEvent.click(
      resendLink,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="resend-error-text"]').textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));

      expect(document.querySelector('.resend-link-text').textContent).toBe(
        `Resend`,
      );
    });
  });

  it('displays the generic error and resend option is enabled after resend otp failed for unknown reason', async () => {
    const otpInput = { phoneNumber };
    const mocks = [
      {
        request: {
          query: AuthMutations.GenerateOTP,
          variables: { input: otpInput },
        },
        error: new Error('An error occurred'),
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );
    jest.runAllTimers();
    const resendLink = document.querySelector('[data-testid="resend-link"]');
    fireEvent.click(
      resendLink,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="resend-error-text"]').textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));

      expect(document.querySelector('.resend-link-text').textContent).toBe(
        `Resend`,
      );
    });
  });

  it('redirects to the prescription details page after a successful OTP verfication', async () => {
    const mocks = [
      {
        request: {
          query: AuthMutations.ValidateOTP,
          variables: { input: otpVerficationInput },
        },
        result: { data: { validateOTP: { success: true, error: null } } },
      },
    ];

    render(
      <BrowserRouter>
        <IntlProvider locale="en">
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} history={mockHistory} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );

    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, {
      target: { value: `${otpVerficationInput.otp}` },
    });

    const verifyButton = document.querySelector('[data-testid="nextButton"]');
    fireEvent.click(
      verifyButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(mockHistory.push).toBeCalledWith({
        // TODO: get the prescriptionId passed here
        pathname: '/prescriptions/123456798',
      });
    });
  });

  it('displays the specific errorvafter validate otp failed with a specific reason', async () => {
    const mocks = [
      {
        request: {
          query: AuthMutations.ValidateOTP,
          variables: { input: otpVerficationInput },
        },
        result: {
          data: {
            validateOTP: {
              success: false,
              error: { code: AuthErrorCode.InvalidCredentials },
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.error.invalidCredentials';

    const messages = {
      [errorMsgCode]: 'Some error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );

    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, {
      target: { value: `${otpVerficationInput.otp}` },
    });

    const verifyButton = document.querySelector('[data-testid="nextButton"]');
    fireEvent.click(
      verifyButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="validation-error-text"]')
          .textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));
    });
  });

  it('displays the generic error after validate otp failed with an unknown reason', async () => {
    const mocks = [
      {
        request: {
          query: AuthMutations.ValidateOTP,
          variables: { input: otpVerficationInput },
        },
        result: {
          data: {
            validateOTP: {
              success: false,
              error: { code: 'auth.err.unknownReason' },
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );

    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, {
      target: { value: `${otpVerficationInput.otp}` },
    });

    const verifyButton = document.querySelector('[data-testid="nextButton"]');
    fireEvent.click(
      verifyButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="validation-error-text"]')
          .textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));
    });
  });

  it('displays the generic error after validate otp failed with missing error code', async () => {
    const mocks = [
      {
        request: {
          query: AuthMutations.ValidateOTP,
          variables: { input: otpVerficationInput },
        },
        result: {
          data: {
            validateOTP: {
              success: false,
              error: {},
            },
          },
        },
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );

    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, {
      target: { value: `${otpVerficationInput.otp}` },
    });

    const verifyButton = document.querySelector('[data-testid="nextButton"]');
    fireEvent.click(
      verifyButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="validation-error-text"]')
          .textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));
    });
  });

  it('displays the generic error after validate otp failed for unknown reason', async () => {
    const mocks = [
      {
        request: {
          query: AuthMutations.ValidateOTP,
          variables: { input: otpVerficationInput },
        },
        error: new Error('An error occurred'),
      },
    ];

    const errorMsgCode = 'phoneLogin.genericAPIError';

    const messages = {
      [errorMsgCode]: 'Sample generic error message',
    };

    render(
      <BrowserRouter>
        <IntlProvider locale="en" messages={messages}>
          <MockedProvider addTypename={false} mocks={mocks}>
            <OTPInput location={mockLocation} />
          </MockedProvider>
        </IntlProvider>
      </BrowserRouter>,
    );

    const input = document.querySelector('.MuiInputBase-input');
    fireEvent.change(input, {
      target: { value: `${otpVerficationInput.otp}` },
    });

    const verifyButton = document.querySelector('[data-testid="nextButton"]');
    fireEvent.click(
      verifyButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    jest.runAllTimers();

    await wait(() => {
      expect(
        document.querySelector('[data-testid="validation-error-text"]')
          .textContent,
      ).toEqual(expect.stringContaining(messages[errorMsgCode]));
    });
  });
});
