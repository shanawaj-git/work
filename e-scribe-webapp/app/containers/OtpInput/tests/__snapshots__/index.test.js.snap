// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OTPInput /> should render and match the snapshot 1`] = `
<div
  class="sc-bdVaJa bIVtPI"
>
  <div>
    <h1
      class="MuiTypography-root MuiTypography-h1 spacing css-qu1isr-MuiTypography-root"
    >
      Enter the code we've sent to you
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 spacing css-1reifv3-MuiTypography-root"
    >
      Enter the 6-digit code we just sent it to +971-xyz-xyzxx-xx
    </p>
    <div
      class="root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 btsVJU otp-input sc-bwzfXH HxnVf css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1fwwy-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
        >
          verification code
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete=""
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            name=""
            placeholder=""
            type="tel"
            value=""
          />
        </div>
      </div>
      
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 inline css-ry76j1-MuiTypography-root"
    >
      Didn't receive code ? 
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 inline css-ry76j1-MuiTypography-root"
      data-testid="counter-value"
    >
      Resend in 60s
    </p>
    
  </div>
  <div
    class="btn-container"
  >
    <button
      class="root StyledButton-sc-1apep4b-0 hPQGaD MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="nextButton"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        style="visibility: visible;"
      >
        Next
      </div>
    </button>
    
  </div>
</div>
`;
